### ISSUE TITLE
Documentation improvement: Addition to Optuna implementation example to bypass redundant calls which may be time intensive

### ISSUE THREAD

**Issue by JLC827** [unit8co/darts#1555]
**Is your feature request related to a current problem? Please describe.**
The documentation provides [a great example of how to implement Optuna for hyperparameter optimisation](https://unit8co.github.io/darts/userguide/hyperparameter_optimization.html)! However, the example relies on the using generating a metric for Optuna using a separate model.predict call. The example assumes that the user wants to make a single model.fit prediction with a forecast horizon across the entire validation set at once. If the user wants to use a validation metric from the model.fit method, it is unclear how they would do so.

(To elaborate on why the user might want to do this, for my situation I do not want to use a single model.predict call as I am using a model with a forecast_horizon of 1, and therefore to produce validation metrics I need to use model.backtest (starting at the validation set date, and with retrain=False). But this method is not parallelised so takes a significant amount of time to run over all my validation data. Hence I'd rather just use the validation metrics generated during training (as I think these are essentially running backtesting anyway?)

**Describe proposed solution**
@dennisbader has already [proposed most of a solution for this in response to another issue. ](https://github.com/unit8co/darts/issues/811). You can then return an Optuna metric by just selecting the best validation metric.

I suggest that this be added to the hyperparameter optimisation docs page or even just add a link to the github issue suggesting that the user use that process if they wish to access validation metrics rather than generating new ones.

**Describe potential alternatives**
Alternatively, I may just be misunderstanding something. If it is explicitly necessary to generate a separate validation metric using model.predict at the end of training please let me know! Or maybe there is a simple way to get model.backtest to run much faster or is that the wrong method to be using?

**Additional context**
Thanks for the library it has saved me a significant amount of time already. This is just a little hurdle which has taken me a while to solve so I think it would be useful to note to prevent new users from hitting the same stumbling block.

**Comment by dennisbader** [comment]
You are right @JLC827 , you can also use the validation losses (or metrics if given) directly to pick the best performing model. 

I added a ticket for a dedicated user guide section on callbacks with torch forecasting models (#1576). 

I would prefer not to make the hyperparameter optimization example too big, but rather reference to the callbacks section from there.
